jupyter_dashboards/nbextension/notebook/dashboard-view/dashboard-actions.js

```diff
@@ -2,131 +2,139 @@
-        $('#urth-dashboard-auth').toggleClass('selected', newState === STATE_DASHBOARD_AUTH);
-        $('#urth-dashboard-view').toggleClass('selected', newState === STATE_DASHBOARD_VIEW);
-        $('#urth-dashboard-show-all, #urth-dashboard-show-all-stacked, #urth-dashboard-hide-all').toggleClass('disabled', newState === STATE_DASHBOARD_VIEW);
-
-        // disable header and toolbar as necessary
-        toggleHeaders(newState === STATE_DASHBOARD_AUTH);
-
-        enableCodeMirror(false);
-
-        enterDbModeCallback(newState === STATE_DASHBOARD_AUTH /* doEnableGrid */);
-        updateUrlState(newState === STATE_DASHBOARD_VIEW);
-
-        // disable scroll to bottom of notebook
-        scrollToBottom = IPython.Notebook.prototype.scroll_to_bottom;
-        IPython.Notebook.prototype.scroll_to_bottom = function() {};
+    function getFaIconClass(el) {
+        return $(el).attr('class').split(' ').filter(function(value) {
+            return /^fa-/.test(value);
+        })[0];
     }
 
-    function setViewButtonEnabled(state) {
-        var idx = VIEW_BTN_IDX[state];
-        $('#urth-dashboard-view-toolbar-buttons > button:nth-of-type(' + idx + ')')
-            .addClass('active')
-            .siblings().removeClass('active');
+    function updateAuthoringBtn(state) {
+        if (state &&
+            state !== STATE.NOTEBOOK &&
+            state !== STATE.DASHBOARD_PREVIEW) {
+            var $btns = $(toolbarBtnsSelector);
+            var icon = getFaIconClass($btns.find(
+                '.dashboard-layout-menu-item[data-dashboard-state="' + state + '"] i'));
+            $btns.find('.dashboard-authoring-btn')
+                .attr('data-dashboard-state', state)
+                .find('.dashboard-authoring-icon')
+                    .removeClass(function() {
+                        return getFaIconClass(this);
+                    })
+                    .addClass(icon);
+        }
     }
 
-    var isHeaderVisible = true;
-    var isToolbarVisible = true;
+    function setDashboardState(newState) {
+        if (newState !== currentState) {
+            if (newState === STATE.NOTEBOOK) {
+                exitDashboardMode();
+            } else {
+                enterDashboardMode(newState);
+            }
+            updateUIState(newState);
+            currentState = newState;
+        }
+    }
 
-    function toggleHeaders(doShow) {
+    function setHeaderVisibility(doShow) {
         // if hiding, save current state
         if (!doShow) {
             isHeaderVisible = $('#header-container').is(':visible');
             isToolbarVisible = $('div#maintoolbar').is(':visible');
         }
-        // hide or revert back to previous saved state
-        $('#header-container').toggle(doShow && isHeaderVisible);
-        $('.header-bar').toggle(doShow && isHeaderVisible);
+        // hide or revert back to previous state
+        $('#header-container, .header-bar').toggle(doShow && isHeaderVisible);
```

Comment:
Logic fails when loading the page with `?dashboard` and then trying to switch to notebook. `doShow=true && isHeaderVisible=false &rarr; false` so the toggle doesn't happen.